File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        ans = []
        currStart, currEnd = intervals[0]
        for i in range(1, len(intervals)):
            newStart, newEnd = intervals[i]
            if newStart <= currEnd:
                currEnd = max(currEnd, newEnd)
            else:
                ans.append([currStart, currEnd])
                currStart, currEnd = newStart, newEnd
        ans.append([currStart, currEnd])
        return ans
        

File: 0056-merge-intervals/0056-merge-intervals.go
func merge(intervals [][]int) [][]int {
  sort.Slice(intervals, func(i, j int) bool {
    return intervals[i][0] < intervals[j][0]
  })
  ans, currStart, currEnd := [][]int{}, intervals[0][0], intervals[0][1]
  for i:=1;i<len(intervals);i++ {
    newStart, newEnd := intervals[i][0], intervals[i][1]
    if newStart <= currEnd {
      currEnd = max(currEnd, newEnd)
    } else {
      ans = append(ans, []int{currStart, currEnd})
      currStart, currEnd = newStart, newEnd
    }
  }
  ans = append(ans, []int{currStart, currEnd})
  return ans
}